src/types.rs

```diff
@@ -148,6 +148,24 @@ pub struct ExternalSourceConfigBase {
     pub is_public: bool,
 }
 
+/// Common attributes to make the symbolicator skip/consider sources by certain criteria.
+#[derive(Deserialize, Clone, Debug, Default)]
+#[serde(default)]
+pub struct SourceFilters {
+    /// File types that are supported by this server.
+    pub filetypes: Vec<FileType>,
+
+    /// When nonempty, a list of prefixes to fuzzy-match filepaths against. The source is then only
+    /// used if one of the path prefixes matches.
+    ///
+    /// "Fuzzy" in this context means that (ascii) casing is ignored, `\` is treated as equal
+    /// to `/` and consecutive slashes are treated as single slash.
+    ///
+    /// If a debug image does not contain any path information it will be treated like an image
+    /// whose path doesn't match any prefix.
+    pub path_prefixes: Vec<String>,
```

Comment:
Let's make these globs instead of just prefixes. Apart from greater flexibility (and efficient library support!) this solves one important issue: With exact prefixes, we cannot specify an arbitrary drive letter for windows paths (e.g. the system could be in `D:\Windows`.

Reply:
Good point, done.

Ironically `glob` conflates `//` with `/` because it assumes a Unix FS. So specifying `C://Windows` as a glob does not seem to match anything.